### PR TITLE
Add prior material amount check to turbine part upgrade

### PR DESCRIPTION
**Pull request by mogeoko**
## About The Pull Request

Currently, upgrading turbine parts will use materials every time the user clicks on it while the upgrade process is active.
The user will now spend a certain amount of materials regardless of the number of clicks.
## Why It's Good For The Game

It doesn't make sense to spend more materials for the same result.
## Changelog
:cl: mogeoko
fix: Turbine parts will now use an amount of materials no greater than needed for the upgrade
/:cl:

### CODE REVIEW

File: code/modules/power/turbine/turbine_parts.dm [hunk 1]
```diff
@@ -64,7 +64,8 @@
 			if(!istype(attacking_item, second_tier_material))
 				return
 			var/obj/item/stack/sheet/second_tier = attacking_item
-			if(second_tier.use(second_tier_material_amount) && do_after(user, 1 SECONDS, src))
+			if(second_tier.get_amount() >= second_tier_material_amount && do_after(user, 1 SECONDS, src))
+				second_tier.use(second_tier_material_amount)
```
Comment (SyncIt21) [inline code/modules/power/turbine/turbine_parts.dm:68]:
```suggestion
			if(do_after(user, 1 SECONDS, src) && second_tier.use(second_tier_material_amount))		
```

We first check if the 1 second has passed and only then use the amount simple.

File: code/modules/power/turbine/turbine_parts.dm [hunk 2]
```diff
@@ -74,7 +75,8 @@
 			if(!istype(attacking_item, third_tier_material))
 				return
 			var/obj/item/stack/sheet/third_tier = attacking_item
-			if(third_tier.use(third_tier_material_amount) && do_after(user, 2 SECONDS, src))
+			if(third_tier.get_amount() >= third_tier_material_amount && do_after(user, 2 SECONDS, src))
+				third_tier.use(third_tier_material_amount)
```
Comment (SyncIt21) [inline code/modules/power/turbine/turbine_parts.dm:79]:
```suggestion
			if(do_after(user, 2 SECONDS, src) && third_tier.use(third_tier_material_amount))			
```

Ditto

File: code/modules/power/turbine/turbine_parts.dm [hunk 3]
```diff
@@ -84,7 +86,8 @@
 			if(!istype(attacking_item, fourth_tier_material))
 				return
 			var/obj/item/stack/sheet/fourth_tier = attacking_item
-			if(fourth_tier.use(fourth_tier_material_amount) && do_after(user, 3 SECONDS, src))
+			if(fourth_tier.get_amount() >= fourth_tier_material_amount && do_after(user, 3 SECONDS, src))
+				fourth_tier.use(fourth_tier_material_amount)
```
Comment (SyncIt21) [inline code/modules/power/turbine/turbine_parts.dm:90]:
```suggestion
			if(do_after(user, 3 SECONDS, src) && fourth_tier.use(fourth_tier_material_amount))		
```

You get it

**Review by SyncIt21 — COMMENTED** [review]
The solution is to actually reverse the order of checks

**Comment by mogeoko** [comment]
Changed the code like you said, many thanks

**Review by Ghommie — APPROVED** [review]

**Pull request merged**